src/lib/storage/run_length_segment/run_length_segment_iterable.hpp

```diff
@@ -17,27 +17,33 @@ class RunLengthSegmentIterable : public PointAccessibleSegmentIterable<RunLength
 
   template <typename Functor>
   void _on_with_iterators(const Functor& functor) const {
+    _segment.access_statistics().on_iterator_create(_segment.size());
     auto begin =
-        Iterator{_segment.values()->cbegin(), _segment.null_values()->cbegin(), _segment.end_positions()->cbegin(), 0u};
+        Iterator{_segment.values()->cbegin(), _segment.null_values()->cbegin(), _segment.end_positions()->cbegin(), 0u,
+                 &_segment};
```

Comment:
Passing both the segment and information that could also be retrieved from the segment is weird. Could we also pass just the statistics object?

Reply:
Since we don't need on_iterator_dereference anymore, we don't need to pass the segment.